packages/appcd-nodejs/src/nodejs.js

```diff
@@ -386,66 +402,157 @@ export function spawnNode({ arch, args, detached, nodeHome, nodeArgs, stdio, v8m
 		throw new Error('Expected arch to be "x86" or "x64"');
 	}
 
-	return Promise.resolve()
-		.then(() => prepareNode({ arch, nodeHome, version }))
-		.then(node => {
-			if (!Array.isArray(nodeArgs)) {
-				nodeArgs = [];
-			}
+	const node = await prepareNode({ arch, nodeHome, version });
+	if (!Array.isArray(nodeArgs)) {
+		nodeArgs = [];
+	}
+
+	if (v8mem && !nodeArgs.some(arg => arg.indexOf('--max_old_space_size=') === 0)) {
+		const arg = generateV8MemoryArgument(v8mem, arch);
+		if (arg) {
+			nodeArgs.push(arg);
+		}
+	}
+
+	args.unshift.apply(args, nodeArgs);
 
-			if (v8mem && !nodeArgs.some(arg => arg.indexOf('--max_old_space_size=') === 0)) {
-				const arg = generateV8MemoryArgument(v8mem, arch);
-				if (arg) {
-					nodeArgs.push(arg);
+	const opts = {
+		stdio: 'inherit'
+	};
+
+	if (detached) {
+		opts.detached    = true;
+		opts.stdio       = 'ignore';
+		opts.windowsHide = true;
+	}
+
+	if (stdio) {
+		// if stdio is set, then override the default
+		opts.stdio = stdio;
+	}
+
+	const prettyArgs = args
+		.map(s => {
+			return s.indexOf(' ') === -1 ? s : `"${s}"`;
+		})
+		.join(' ');
+
+	logger.log('Spawning: %s', highlight(`${node} ${prettyArgs} # ${JSON.stringify(opts)}`));
+
+	// write the last run file
+	fs.writeFileSync(path.join(path.dirname(node), '.lastrun'), Date.now());
+
+	let tries = 3;
+
+	return (function trySpawn() {
+		return Promise.resolve()
+			.then(() => {
+				tries--;
+				const child = spawn(node, args, opts);
+				if (detached) {
+					child.unref();
 				}
-			}
+				return child;
+			})
+			.catch(err => {
+				if ((err.code === 'ETXTBSY' || err.code === 'EBUSY') && tries) {
+					logger.log(`Spawn threw ${err.code}, retrying...`);
+					return new Promise((resolve, reject) => {
+						setTimeout(() => trySpawn().then(resolve, reject), 50);
+					});
+				}
+				throw err;
+			});
+	}());
+}
+
+/**
+ * Finds all unused Node.js executables that are older than the specified max age and deletes them.
+ *
+ * @param {Number} maxAge - The maximum age for unused Node.js executables.
+ * @returns {Array.<String>} A list of unused Node.js executables that were removed.
+ */
+export function purgeUnusedNodejsExecutables({ maxAge, nodeHome }) {
+	const purged = [];
+
+	if (!isDir(nodeHome)) {
+		return purged;
+	}
 
-			args.unshift.apply(args, nodeArgs);
+	const nodeExecutable = getNodeFilename();
+	const now = Date.now();
 
-			const opts = {
-				stdio: 'inherit'
-			};
+	const isEmpty = dir => {
+		return fs.readdirSync(dir).filter(name => name === '.DS_Store').length > 0;
```

Comment:
`return fs.readdirSync(dir).filter(name => name !== '.DS_Store').length === 0;`? Currently seeing executables removed if the folders contain a .DS_Store file